releasetool/data/roles/download_rpms/tasks/download.yaml

```diff
@@ -0,0 +1,21 @@
+---
+- name: list {{ os_version }} RPMs from Koji
+  command: koji list-tagged --latest --quiet --inherit --sigs --arch=src {{ item }}
+  register: koji_list
+  loop: "{{ koji_tags }}"
+
+- name: filter koji result
+  set_fact:
+    koji_list_filtered: "{{ koji_list.results | map(attribute='stdout_lines') | flatten | map('regex_replace', '\\.src$') | list }}"
+
+- name: select rpms from the koji output
+  set_fact:
+    all_rpms: "{{ koji_list_filtered | map('regex_replace', '^.* ') | list | unique }}"
+    signed_rpms: "{{ koji_list_filtered | select('match', gpg_id~' ') | map('regex_replace', '^.* ') | list | unique }}"
+
+- name: download rpms
+  command: koji download-build --debuginfo {{ item }}
+  args:
+    chdir: "{{ work_dir_rpm }}"
+  loop: "{{ all_rpms | difference(signed_rpms) }}"
```

Comment:
Do you see any value in using Koji as a python library and creating modules?

Reply:
long term, yes, but the koji library is pretty weird, so writing own modules would require some ramp up.